### DISCUSSION TITLE
What exactly is --pack-dyn-relocs linker flag and how does it help reduce library size?

### DISCUSSION THREAD

**Discussion by pendyalasyam** [android/ndk#1831] [Q&A] [answered]
In NDK Build System Maintainers Guide, it is [mentioned](https://android.googlesource.com/platform/ndk/+/master/docs/BuildSystemMaintainers.md#Controlling-Binary-Size:%7E:text=Beginning%20with%20API%20level%2023%20it%20is%20possible%20to%20compress%20the%20relation%20data%20in%20libraries%20and%20executables.%20Libraries%20with%20large%20numbers%20of%20relocations%20will%20benefit%20from%20this.%20Enable%20with%20%2DWl%2C%2D%2Dpack%2Ddyn%2Drelocs%3Dandroid%20at%20link%20time.) that -Wl,--pack-dyn-relocs=android is useful to control the size of libraries that have large number of relocations.

Could somebody please help me understand

What are dynamic relocations and how are they used [Any links to videos or tutorials will be helpful] ?
What is "pack" in --pack-dyn-relocs? Is it some kind of using shortcuts to reduce the binary size?
How does -Wl,--pack-dyn-relocs=android help to reduce the library size?
How to check which exact section of library is getting effected with and without using this flag.
Thanks!!

**Comment by enh-google** [comment, marked as answer]
https://maskray.me/blog/2021-10-31-relative-relocations-and-relr
